dotnet/src/Agents/Abstractions/AgentThread.cs

```diff
@@ -26,6 +27,52 @@ public abstract class AgentThread
     /// </summary>
     public virtual bool IsDeleted { get; protected set; } = false;
 
+    /// <summary>
+    /// Gets or sets the container for <see cref="AIContextBehavior"/> objects that manages their lifecycle and interactions.
+    /// </summary>
+    [Experimental("SKEXP0110")]
+    public virtual AIContextBehaviorsManager AIContextBehaviors { get; init; } = new AIContextBehaviorsManager();
```

Comment:
We may be a bit far from the naming discussion, but I would read it better if it was:

`AIBehaviorContext` sounds best IMHO.


```suggestion
    public virtual AIBehaviorContextManager BehaviorContexts { get; init; } = new();
```

Reply:
This isn't a context though, it's a behavior.  If it ends in context, it would make it sound like a context, not a behavior attached to the context.